src/main/java/org/spongepowered/common/command/SpongeCommandFactory.java

```diff
@@ -560,9 +561,15 @@ private static CommandSpec createSpongePluginsCommand() {
                 .arguments(optionalWeak(literal(Text.of("reload"), "reload")), optional(plugin(Text.of("plugin"))))
                 .executor((src, args) -> {
                     if (args.hasAny("reload") && src.hasPermission("sponge.command.plugins.reload")) {
-                        src.sendMessage(Text.of("Sending reload event to all plugins. Please wait."));
                         Sponge.getCauseStackManager().pushCause(src);
-                        SpongeImpl.postEvent(SpongeEventFactory.createGameReloadEvent(Sponge.getCauseStackManager().getCurrentCause()));
+                        if (args.hasAny("plugin")) {
+                            PluginContainer plugin = args.<PluginContainer>getOne("plugin").get();
+                            src.sendMessage(Text.of("Sending reload event to " + plugin.getId() + ". Please wait."));
+                            ((SpongeEventManager) Sponge.getEventManager()).post(SpongeEventFactory.createGameReloadEvent(Sponge.getCauseStackManager().getCurrentCause()), plugin);
+                        } else {
+                            src.sendMessage(Text.of("Sending reload event to all plugins. Please wait."));
+                            SpongeImpl.postEvent(SpongeEventFactory.createGameReloadEvent(Sponge.getCauseStackManager().getCurrentCause()));
```

Comment:
Could you potentially provide a test plugin to verify the causes used for this event in this case?

Reply:
No, sorry. I don't know enough about it or what I would even do to test that. The calls to push/pop are in the same places so there shouldn't be a difference here.

Reply:
Write a plugin in the [testplugins directory](https://github.com/SpongePowered/SpongeCommon/tree/bleeding/testplugins/src/main/java/org/spongepowered/test) just as you would, to implement the listener and maybe just print out the cause from the event? 
```java
@Listener
public void onReload(GameReloadEvent event) {
  this.logger.log(event.getCause());
}
```

Reply:
What would be the point of this? It doesn't seem like there's actually anything that would really be being tested here.

Reply:
Real world testing. What it allows us to do is run the server in a test scenario and check things are working with a known set of plugins easily in the IDE. In this case, have the reload event in one allows us to check that the reload event fires as expected, to all or to a target, and makes sure nothing has regressed.

Reply:
Gotcha, I wasn't thinking of testing the event being sent to specifically the test plugin and was mainly confused at what the point of just logging the cause would be as it hasn't changed. That makes a lot more sense now and I can add that in.